src/ai/backend/gateway/vfolder.py

```diff
@@ -1048,7 +1119,7 @@ async def download_with_token(request) -> web.StreamResponse:
     except (ValueError, FileNotFoundError):
         raise InvalidAPIParameters('The file is not found.')
     if not file_path.is_file():
-        raise InvalidAPIParameters('The file is not a regular file.')
+        return await download_directory_as_archive(request, file_path)
```

Comment:
Same here.

Reply:
I added `archive: boolean` parameter for downloading directory as an archive.